### PR TITLE
Add Bazel build rules

### PR DESCRIPTION
**Pull request by alexeagle**
They are not used yet.

### CODE REVIEW

File: .bazelrc [hunk 1]
```diff
@@ -0,0 +1,2 @@
+# Performance: avoid stat'ing input files
+build --watchfs
```
Comment (evmar) [inline .bazelrc]: Does this matter?  tsickle is tiny and stat is fast.

Comment (evmar) [inline .bazelrc]: https://github.com/ninja-build/ninja/wiki/Timing-Numbers

File: .circleci/config.yml [hunk 2]
```diff
@@ -0,0 +1,16 @@
+version: 2
+jobs:
+  build:
+    working_directory: ~/tsickle
+    docker:
+      - image: alexeagle/ngcontainer
+    steps:
+      - checkout
+      - restore_cache:
+          key: tsickle-{{ .Branch }}-{{ checksum "yarn.lock" }}
```
Comment (evmar) [inline .circleci/config.yml]: sha1sum instead of checksum (?)
I think checksums aren't intended to be used for for content id

File: .gitignore [hunk 3]
```diff
@@ -1,5 +1,5 @@
 # Compiled JS
-build
+/built
```
Comment (evmar) [inline .gitignore]: Why change this?

File: WORKSPACE [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+
+load("@io_bazel_rules_typescript//:defs.bzl", "node_repositories", "yarn_install")
+
+# Install a hermetic version of node
+# After this is run, a label @io_bazel_typescript_node//:bin/node will exist
+node_repositories()
+
+# Install yarn, and run yarn install to create node_modules.
+# Also creates the BUILD file under node_modules.
+yarn_install(package_json = "//:package.json")
```
Comment (evmar) [inline WORKSPACE]: Can this be :package.json or do you need the full path?

File: src/BUILD [hunk 5]
```diff
@@ -0,0 +1,9 @@
+package(default_visibility=["//visibility:public"])
+load("@io_bazel_rules_typescript//:defs.bzl", "ts_library")
+
+ts_library(
+    name = "src",
+    srcs = glob(["*.ts"]),
```
Comment (evmar) [inline src/BUILD]: shouldn't this exclude tests?  or we have all tests in a different dir?

File: src/BUILD [hunk 6]
```diff
@@ -0,0 +1,9 @@
...
+    deps = [],
+    tsconfig = "//:tsconfig.json",
```
Comment (evmar) [inline src/BUILD]: weird to see library-specific tsconfigs!

File: .bazelrc (hunk 1 above)
Comment (alexeagle) [inline .bazelrc]: No I was just cargo-culting from the Angular setup I'm working on, removed.

File: .circleci/config.yml (hunk 2 above)
Comment (alexeagle) [inline .circleci/config.yml]: It's a SHA256: https://circleci.com/docs/2.0/configuration-reference/#save_cache
(it's not an arbitrary unix command, it's part of their config DSL)

File: .gitignore (hunk 3 above)
Comment (alexeagle) [inline .gitignore]: surprising things happen on Mac when you have both BUILD and build/
surprising in a bad way

File: WORKSPACE (hunk 4 above)
Comment (alexeagle) [inline WORKSPACE]: it totally can, since there's a BUILD file in this directory, it's in the current package

File: src/BUILD (hunk 6 above)
Comment (alexeagle) [inline src/BUILD]: yeah, it's different

File: src/BUILD (hunk 5 above)
Comment (alexeagle) [inline src/BUILD]: tests are in /test/

File: .gitignore (hunk 3 above)
Comment (evmar) [inline .gitignore]: hahahaha

**Pull request merged**